src/msw/notebook.cpp

```diff
@@ -729,22 +730,195 @@ bool wxNotebook::InsertPage(size_t nPage,
     return true;
 }
 
+bool wxNotebook::MSWHitTestLeftRight(const wxPoint &pt, int item, long *flags) const
+{
+    // In Windows there is a bug so that the flags are not properly reported
+    // for left/right orientation, so we have to do it manually.
+    // It seems that the tabs are evaluated the same as if they were top/bottom
+    // oriented, because if multiple tabs are stacked the flags are quite different
+    // for each tab on the x axis.
+    wxDirection tabDir = GetTabOrientation();
+    if (!(tabDir == wxLEFT || tabDir == wxRIGHT))
+        return false;
+
+    long dummy = 0;
+    if (!flags)
+        flags = &dummy;
+
+    wxRect r = GetRect();
+    if (!r.Contains(pt))
+    {
+        *flags = wxBK_HITTEST_NOWHERE;
+        return true;
+    }
+
+    size_t pages = GetPageCount();
+    if (!pages)
+    {
+        *flags |= wxBK_HITTEST_NOWHERE | wxBK_HITTEST_ONPAGE;
+        return true;
+    }
+
+    if (GetPageSize().Contains(pt))
+    {
+        *flags = wxBK_HITTEST_ONPAGE;
+        return true;
+    }
+
+    r = GetTabRect(item);
+    if (!r.Contains(pt))
+    {
+        *flags = wxBK_HITTEST_NOWHERE;
+        return true;
+    }
+
+    // Just a magic number between the icon and the start of the text
+    // I checked with different resolutions and it seems to be about it. Have to check with
+    // different scalings as well.
+    int gap = 8;
+
+    TCHAR buffer[256];
+    TCITEM ti;
+    ti.mask = TCIF_IMAGE|TCIF_TEXT;
+    ti.cchTextMax = sizeof(buffer) - 1;
+    ti.pszText = buffer;
+    TabCtrl_GetItem(GetHwnd(), item, &ti);
+
+    wxRect tr;
+    wxRect ir;
+
+    // If the tab has an image, we need to check if the mouse is over it.
+    if (ti.iImage != -1)
+    {
+        wxIcon const &icon = GetImage(ti.iImage);
+        if (!icon.IsNull())
```

Comment:
It's better to avoid double negation:
```suggestion
        if (icon.IsOk())
```